### PR TITLE
Preliminary PR with documentation of Adapters

### PR DESCRIPTION
**Pull request by awholegunch**
Users' Manual documentation

Staged at: [Users' Manual](http://stg-web101.sw.joyent.continuuity.net/cdap/3.0.0-SNAPSHOT-r3.0_users_manual/en/users-manual/index.html)

### CODE REVIEW

File: cdap-docs/users-manual/source/etl/index.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+============
+ETL Overview
+============
+
+
+What is ETL?
+============
+ETL is **Extract**, **Transformation** and **Loading** of data, and is a common first-step
+in any data application. CDAP endeavors to make performing ETL possible out-of-box without
+writing code; instead, you just configure an ETL Adaptor and then operate it.
```
Comment (gokulavasan) [inline cdap-docs/users-manual/source/etl/index.rst]: Adaptor -> Adapter. Change other instances too.

**Comment by sreevatsanraman** [comment]
Adapter not Adaptor everywhere.

File: cdap-docs/developers-manual/source/advanced/etl.rst [hunk 2]
```diff
@@ -0,0 +1,85 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _advanced-custom-etl:
+
+===============================================
+Creating Custom ETL Applications and Components
```
Comment (gokulavasan) [inline cdap-docs/developers-manual/source/advanced/etl.rst]: What are ETL Applications? Are you referring to creating ETL Plugins? 

It should be called 
"Creating Custom ETL Plugins"

Comment (gokulavasan) [inline cdap-docs/developers-manual/source/advanced/etl.rst]: I see we have "Creating Application Templates" also below. So we should mention that this page, documents both. Or we can split it into two pages.

File: cdap-docs/developers-manual/source/advanced/etl.rst [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+===============================================
+
+Overview
+========
+- Purpose: For people writing new ETL Applications and ETL Components
```
Comment (gokulavasan) [inline cdap-docs/developers-manual/source/advanced/etl.rst]: We are not going to use ETL Applications, ETL Components.

ETL Adapters. ETL Plugins.

File: cdap-docs/reference-manual/source/glossary.rst [hunk 4]
```diff
@@ -145,3 +148,72 @@ Glossary
       the data model and the schema that represents the data are optimized for querying
       and aggregating over time ranges.
 
+   .. ETL and Application Templates
+
```
Comment (gokulavasan) [inline cdap-docs/reference-manual/source/glossary.rst]: The ETL parts of glossary section needs to be revisited.

File: cdap-docs/users-manual/source/etl/creating.rst [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+
+
+Examples
+--------
+
+::
+
+  # Batch
+
+  # Retrieving the available types of plugins of the ETL Batch Application Template
```
Comment (gokulavasan) [inline cdap-docs/users-manual/source/etl/creating.rst]: I think we should format the output through " python -m json.tool " to get better formatting.

File: cdap-docs/users-manual/source/etl/creating.rst [hunk 6]
```diff
@@ -0,0 +1,422 @@
...
+  a String. If unspecified, the 'body' key is
+  used.","type":"string","required":false}},"template":{"name":"etlRealtime","description":"
+  Realtime Extract-Transform-Load (ETL)
+  Adapter","programType":"Worker"},"source":{"fileName":"cdap-etl-lib-3.0.0-SNAPSHOT.jar","
+  name":"cdap-etl-lib","version":{"version":"3.0.0-SNAPSHOT","major":3,"minor":0,"fix":0,"
+  suffix":"SNAPSHOT"}},"type":"sink","name":"Stream","description":"Realtime sink that
+  outputs to the specified CDAP Stream"}]
+
+
+Creating and Deploying an Adapter
```
Comment (gokulavasan) [inline cdap-docs/users-manual/source/etl/creating.rst]: Lets say, Creating an Adapter. There is no "deploying of an adapter"

File: cdap-docs/developers-manual/source/advanced/atap.rst [hunk 7]
```diff
@@ -0,0 +1,203 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _advanced-custom-etl:
+
+==============================
+Creating Application Templates
```
Comment (gokulavasan) [inline cdap-docs/developers-manual/source/advanced/atap.rst]: Can we update the staging server. I don't see this page on it.

File: cdap-docs/developers-manual/source/advanced/atap.rst [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+Written in Javascript?
+
+
+Creating Plugins
+=================
+
+Overview
+--------
+- Written in Java
+- Accessing existing Servers and Sinks
```
Comment (gokulavasan) [inline cdap-docs/developers-manual/source/advanced/atap.rst]: What is meant by Accessing existing Servers (Sources?) and Sinks?

**Comment by awholegunch** [comment]
This work has changed radically and many of the comments no longer apply. Will reopen when ready.

**Pull request closed without merging**